### DISCUSSION TITLE
about the old function  "CreateSinglerObject"

### DISCUSSION THREAD

**Discussion by hhx465453939** [LTLA/SingleR#226] [Q&A]
hi 
this package is really great in SCrna analysis,  but I have met some trouble recently when using the function " CreateSinglerObject", It just reported an error and made cell cluster annotation aborted:

```
singler = classifySingleR(counts, annot = ann, "pbmc", min.genes = 0,
+                               species = "Human", citation = "",
+                               ref.list = list(), normalize.gene.length = F, variable.genes = "de",
+                               fine.tune = F, do.signatures = T, clusters = clusters, do.main.types = T,
+                               reduce.file.size = T, numCores = 1)
Error in classifySingleR(counts, annot = ann, "pbmc", min.genes = 0, species = "Human",  : 
  unused arguments (annot = ann, min.genes = 0, species = "Human", citation = "", ref.list = list(), normalize.gene.length = F, variable.genes = "de", do.signatures = T, clusters = clusters, do.main.types = T, reduce.file.size = T, numCores = 1)
```
I have checked the Bioconductor reference but found nothing about this function if there was any replacement for this function in the latest version. 
this problem really matters to me, and please give me some advice, thank you!

**Comment by LTLA** [comment]
The Bioconductor version of this package has not had a `CreateSinglerObject` function for a very, very long time. In fact, it has never had it, because one of our first steps was to switch to Bioconductor's standardized objects, i.e., the `SummarizedExperiment` and its derivatives.

It seems you are using the original version of the package at https://github.com/dviraran/SingleR. If you want to use the Bioconductor version, take a look at the [SingleR book](http://bioconductor.org/books/release/SingleRBook/) for the new calls.

**Reply by hhx465453939** [reply]
thank you very much for your advice :)
it seems that the original version is not compatible with the R version 4.2.0, I really need to adjust my script referencing to SinglrR book, I'm not a professional in bioinformatics, and your reply is really helpful, thanks again for that.
